File: cli/cmd/me.go
package cmd

import (
	"strconv"
	"time"

	"github.com/spf13/cobra"

	"openpa.local/cli/internal/output"
)

func newMeCmd() *cobra.Command {
	return &cobra.Command{
		Use:   "me",
		Short: "Show identity info for the current OPENPA_TOKEN",
		RunE: func(cmd *cobra.Command, args []string) error {
			if err := requireToken(); err != nil {
				return err
			}
			ctx := cmd.Context()
			me, err := state.Client.Me(ctx)
			if err != nil {
				return err
			}
			if state.Output.JSON {
				return output.PrintJSON(me)
			}
			output.PrintKV([][2]string{
				{"profile", me.Profile},
				{"subject", me.Subject},
				{"issued_at", formatUnix(me.IssuedAt)},
				{"expires_at", formatUnix(me.ExpiresAt)},
				{"working_dir", me.WorkingDir},
				{"user_working_dir", me.UserWorkingDir},
			})
			return nil
		},
	}
}

func formatUnix(ts int64) string {
	if ts == 0 {
		return ""
	}
	t := time.Unix(ts, 0)
	return t.Format(time.RFC3339) + " (" + strconv.FormatInt(ts, 10) + ")"
}

File: app/cli/commands/me.py
"""`openpa me` — show identity info for the current OPENPA_TOKEN.

Mirrors `cli/cmd/me.go`.
"""

from __future__ import annotations

from datetime import datetime, timezone

import typer

from app.cli.commands._helpers import graceful_errors


@graceful_errors
def me(ctx: typer.Context) -> None:
    """Show identity info for the current OPENPA_TOKEN."""
    import asyncio

    from app.cli.client._base import Client
    from app.cli.client.me import get_me
    from app.cli.config import Config
    from app.cli.output import OutputMode, print_json, print_kv

    cfg: Config = ctx.obj["cfg"]
    mode: OutputMode = ctx.obj["mode"]
    cfg.require_token()

    async def _run() -> None:
        async with Client(cfg) as client:
            return await get_me(client)

    me_info = asyncio.run(_run())

    if mode.json:
        print_json(me_info.to_dict())
        return

    print_kv([
        ("profile", me_info.profile),
        ("subject", me_info.subject),
        ("issued_at", _format_unix(me_info.issued_at)),
        ("expires_at", _format_unix(me_info.expires_at)),
        ("working_dir", me_info.working_dir),
        ("user_working_dir", me_info.user_working_dir),
    ])


def _format_unix(ts: int) -> str:
    if ts == 0:
        return ""
    dt = datetime.fromtimestamp(ts, tz=timezone.utc)
    return f"{dt.isoformat(timespec='seconds')} ({ts})"
